File: SaasFeeGuides/SaasFeeGuides/Models/Activity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SaasFeeGuides.Models
{
    public class Activity : IEquatable<Activity>
    {
        public int? Id { get; set; }
        public string Name { get; set; }
        public string TitleContentId { get; set; }
        public string DescriptionContentId { get; set; }
        public string MenuImageContentId { get; set; }
        public string VideoContentId { get; set; }
        public string ImageContentId { get; set; }
        public bool IsActive { get; set; }
        public string CategoryName { get; set; }


        public IList<ActivitySku> Skus { get; set; }
        public IList<ActivityEquiptment> Equiptment { get; set; }

        public override bool Equals(object obj)
        {
            return Equals(obj as Activity);
        }

        public bool Equals(Activity other)
        {
            return other != null &&
                   EqualityComparer<int?>.Default.Equals(Id, other.Id);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Id);
        }
    }
}

File: SaasFeeGuides/SaasFeeGuides/ClientApp/src/app/viewModels/activity.ts

import { Content } from './content';
import { ActivitySku } from './activitySku';

export class Activity {

  constructor(
    public id: number,
    public name: string,
    public titleContent: Content[],
    public descriptionContent: Content[],
    public menuImageContent: Content[],
    public videoContent: Content[],
    public imageContent: Content[],
    public isActive: boolean,
    public categoryName: string,
    public skus: ActivitySku[],
    public equiptment:any[]
    
  ) {
  }
  
}


